### PR TITLE
Add BoardDocs Support

### PR DESCRIPTION
**Pull request by dweekly**
This is a first pass at adding a loader for scraping BoardDocs content, such as is commonly used by public government meetings. It can crawl a list of all the meetings and pull down agenda content; the current iteration doesn't yet support indexing of PDFs linked in the agenda, but that's next on the TODO. :)

### CODE REVIEW

File: loader_hub/boarddocs/base.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    def process_meeting(self,
+                       meeting_id:str,
+                       index_pdfs:bool = True) -> List[Document]:
+        """
+        Returns documents from the given meeting
+        """
+
+        agenda_url = self.base_url + "/PRINT-AgendaDetailed"
+
+        print("Parsing meeting " + meeting_id)
```
Comment (jerryjliu) [inline loader_hub/boarddocs/base.py:77]: remove print statements

File: loader_hub/boarddocs/base.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        import html2text
+        from bs4 import BeautifulSoup
+
+        # parse the returned HTML
+        soup = BeautifulSoup(response.content, "html.parser")
+        agenda_date = soup.find("div", {"class":"print-meeting-date"}).string
+        agenda_title = soup.find("div", {"class":"print-meeting-name"}).string
+        agenda_files = [fd.a.get('href') for fd in soup.find_all("div", {"class":"public-file"})]
+        agenda_data = html2text.html2text(response.text)
+        print(" Agenda Title:", agenda_title)
```
Comment (jerryjliu) [inline loader_hub/boarddocs/base.py:95]: same here

File: loader_hub/boarddocs/base.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        print(" Agenda Date:", agenda_date)
+        # print("Number of Files:",len(agenda_files))
```
Comment (jerryjliu) [inline loader_hub/boarddocs/base.py:97]: remove commented out parts

File: loader_hub/boarddocs/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+To use this loader, you'll need to specify which BoardDocs site you want to load.
+
+```python
+from llama_index import download_loader
+
+BoardDocsReader = download_loader("BoardDocsReader")
+loader = BoardDocsReader(site="ca/redwood", committee_id="A4EP6J588C05")
+# For a site URL https://go.boarddocs.com/ca/redwood/Board.nsf/Public
+# your boarddocssite should be set to 'ca/redwood'
+documents = loader.load_data(boarddocssite=['ca/redwood'])
```
Comment (jerryjliu) [inline loader_hub/boarddocs/README.md:18]: is boarddocssite specified in the code? also why does `ca/redwood` need to be specified twice?

**Review by jerryjliu — COMMENTED** [review]
looks good! some comments

File: loader_hub/boarddocs/README.md (hunk 4 above)
Comment (dweekly) [reply loader_hub/boarddocs/README.md]: Whoops, sorry - I changed the calling patterns but hadn't updated the README. Fixed.

File: loader_hub/boarddocs/base.py (hunk 2 above)
Comment (dweekly) [reply loader_hub/boarddocs/base.py]: Removed.

File: loader_hub/boarddocs/base.py (hunk 1 above)
Comment (dweekly) [reply loader_hub/boarddocs/base.py]: Removed.

**Comment by jerryjliu** [comment]
@dweekly we've fixed the failing tests on `main`, i can help merge it in

**Pull request merged**